packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/ApiKeyAuthImpl.kt

```diff
@@ -0,0 +1,162 @@
+
+    private fun unwrap(apiKeyData: ApiKeyWrapper): ApiKey {
+        return ApiKey(
+            ObjectIdImpl(apiKeyData.id),
+            apiKeyData.value,
+            apiKeyData.name,
+            !apiKeyData.disabled
+        )
+    }
+
+    override suspend fun create(name: String): ApiKey {
+        try {
+            Channel<Result<ApiKey>>(1).use { channel ->
+                RealmInterop.realm_app_user_apikey_provider_client_create_apikey(
+                    app.nativePointer,
+                    user.nativePointer,
+                    name,
+                    channelResultCallback<ApiKeyWrapper, ApiKey>(channel) { apiKeyData ->
+                        unwrap(apiKeyData)
+                    }.freeze()
+                )
+                return channel.receive()
+                    .getOrThrow()
+            }
+        } catch (ex: ServiceException) {
+            if (ex.message?.contains("[Service][InvalidParameter(6)] can only contain ASCII letters, numbers, underscores, and hyphens.") == true ||
+                ex.message?.contains("[Service][Unknown(-1)] 'name' is a required string.") == true
+            ) {
+                throw IllegalArgumentException(ex.message!!)
+            } else {
+                throw ex
+            }
+        }
+    }
+
+    override suspend fun delete(id: ObjectId) {
+        try {
+            Channel<Result<Unit>>(1).use { channel ->
+                RealmInterop.realm_app_user_apikey_provider_client_delete_apikey(
+                    app.nativePointer,
+                    user.nativePointer,
+                    id as ObjectIdWrapper,
+                    channelResultCallback<Unit, Unit>(channel) {
+                        // No-op
+                    }.freeze()
+                )
+                return channel.receive().getOrThrow()
+            }
+        } catch (ex: AppException) {
+            if (ex.message?.contains("[Service][Unknown(-1)] expected Authorization header with JWT (Bearer schema).") == true) {
+                throw ex
+            } else {
+                // No-op
+            }
+        }
+    }
+
+    override suspend fun disable(id: ObjectId) {
+        try {
+            Channel<Result<Unit>>(1).use { channel ->
+                RealmInterop.realm_app_user_apikey_provider_client_disable_apikey(
+                    app.nativePointer,
+                    user.nativePointer,
+                    id as ObjectIdWrapper,
+                    channelResultCallback<Unit, Unit>(channel) {
+                        // No-op
+                    }.freeze()
+                )
+                return channel.receive()
+                    .getOrThrow()
+            }
+        } catch (ex: ServiceException) {
+            if (ex.message?.contains("[Service][ApiKeyNotFound(35)] API key not found.") == true) {
+                throw IllegalArgumentException(ex.message!!)
+            } else {
+                throw ex
+            }
```

Comment:
Same here. If this is written to handle the unexpected behaviour you spotted while disabling the provider in the server you should write a `TODO` comment pointing to the github ticket where you reported that issue.